src/Test.UnitTests.Sarif/Baseline2/WhereComparerTests.cs

```diff
@@ -0,0 +1,190 @@
+﻿using System;
+using System.Collections.Generic;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Sarif.Baseline
+{
+    public class WhereComparerTests
+    {
+        private static readonly Uri Uri1 = new Uri("https://github.com/microsoft/sarif-sdk/blob/master/src/Sarif/Errors.cs");
+        private static readonly Uri Uri2 = new Uri("https://github.com/microsoft/sarif-sdk/blob/master/src/Sarif/Notes.cs");
+
+        private static readonly Region SampleRegion = new Region() { StartLine = 10, StartColumn = 5, EndLine = 11, EndColumn = 9 };
+
+        private static readonly ArtifactLocation ArtifactLocationUri1 = new ArtifactLocation() { Uri = Uri1 };
+        private static readonly ArtifactLocation ArtifactLocationUri2 = new ArtifactLocation() { Uri = Uri2 };
+
+        private static readonly LogicalLocation LogicalLocation1 = new LogicalLocation() { FullyQualifiedName = "Namespace1.Class1.Method1" };
+        private static readonly LogicalLocation LogicalLocation2 = new LogicalLocation() { FullyQualifiedName = "Namespace1.Class1.Method2" };
+
+        [Fact]
+        public void WhereComparer_Region()
+        {
+            // Reminder: CompareTo less than zero means left sorts before right.
+            Assert.True(5.CompareTo(10) < 0);
+
+            Region left = new Region(SampleRegion);
+            Region right;
+
+            // Equal
+            right = new Region(left);
+            Assert.True(WhereComparer.CompareTo(left, right) == 0);
+
+            // Right StartLine > Left StartLine
+            right = new Region(left) { StartLine = left.StartLine + 1 };
+            Assert.True(WhereComparer.CompareTo(left, right) < 0);
+
+            // Right StartLine > Left StartLine, lower precedence stuff less
+            right = new Region(left) { StartLine = left.StartLine + 1, StartColumn = left.StartColumn - 1, EndLine = left.EndLine - 1, EndColumn = left.EndColumn - 1 };
+            Assert.True(WhereComparer.CompareTo(left, right) < 0);
+
+            // Right StartColumn > Left StartColumn
+            right = new Region(left) { StartColumn = left.StartColumn + 1, EndLine = left.EndLine - 1, EndColumn = left.EndColumn - 1 };
+            Assert.True(WhereComparer.CompareTo(left, right) < 0);
+
+            // Right EndLine > Left EndLine
+            right = new Region(left) { EndLine = left.EndLine + 1, EndColumn = left.EndColumn - 1 };
+            Assert.True(WhereComparer.CompareTo(left, right) < 0);
+
+            // Right EndColumn > Left EndColumn
+            right = new Region(left) { EndColumn = left.EndColumn + 1 };
+            Assert.True(WhereComparer.CompareTo(left, right) < 0);
+
+            left.ByteOffset = 100;
+            left.ByteLength = 10;
+
+            // Right Offset bigger, supercedes others
```

Comment:
>supercedes [](start = 36, length = 10)

It's not really a matter of superseding. The spec distinguishes three types of region property sets:
- line/column properties
- character offset/length properties
- byte offset/length properties

At least one set must be present. If more than one set is present, the **MUST**  refer to exactly the same range of bytes in the file. So you are entitled to look at any set you wish, and if another set gives a different answer, that's just too bad; the SARIF file is invalid.

So here, you are perfectly entitled to look at the byte offset/length properties to decide which region is "smaller", but the fact that this works, given that your test `Region` object is inconsistent, is an artifact of the fact that you happen to look at those properties first.

As long as you keep your implementation the same there's nothing wrong with this.